js/node/src/inference_session_wrap.cc

```diff
@@ -10,13 +10,16 @@
 #include "tensor_helper.h"
 
 Napi::FunctionReference InferenceSessionWrap::constructor;
-Ort::Env *InferenceSessionWrap::ortEnv;
 
 Napi::Object InferenceSessionWrap::Init(Napi::Env env, Napi::Object exports) {
   // create ONNX runtime env
   Ort::InitApi();
-  ortEnv = new Ort::Env{ORT_LOGGING_LEVEL_WARNING, "onnxruntime-node"};
-
+  ORT_NAPI_THROW_ERROR_IF(
+      Ort::Global<void>::api_ == nullptr, env,
+      "Failed to initialize ONNX Runtime API. It could happen when this nodejs binding was built with a higher version "
+      "ONNX Runtime but now runs with a lower version ONNX Runtime DLL(or shared library).");
+  auto ortEnv = new Ort::Env{ORT_LOGGING_LEVEL_WARNING, "onnxruntime-node"};
+  env.SetInstanceData(ortEnv);
```

Comment:
Does `Napi::Env::SetInstanceData()` requires NAPI v6?

when is this ortEnv get released?

Reply:
Yes. it is.

Reply:
It get released when  NodeMainInstance::Run() returns. https://github.com/nodejs/node/blob/main/src/node_main_instance.cc#L77